packages/twenty-front/src/modules/workflow/components/WorkflowFormTextFieldInput.tsx

```diff
@@ -0,0 +1,59 @@
+import { TextVariableEditor } from '@/object-record/record-field/form-types/components/TextVariableEditor';
+import { useTextVariableEditor } from '@/object-record/record-field/form-types/hooks/useTextVariableEditor';
+import { WorkflowFormFieldInputBase } from '@/workflow/components/WorkflowFormFieldInputBase';
+import { parseEditorContent } from '@/workflow/search-variables/utils/parseEditorContent';
+import { isDefined } from 'twenty-ui';
+
+type WorkflowFormTextFieldInputProps = {
+  label?: string;
+  defaultValue: string | undefined;
+  placeholder: string;
+  onPersist: (value: null | string) => void;
+  multiline?: boolean;
+  readonly?: boolean;
+};
+
+export const WorkflowFormTextFieldInput = ({
+  label,
+  defaultValue,
+  placeholder,
+  onPersist,
+  multiline,
+  readonly,
+}: WorkflowFormTextFieldInputProps) => {
+  const editor = useTextVariableEditor({
+    placeholder,
+    multiline,
+    readonly,
+    defaultValue,
+    onUpdate: (editor) => {
+      const jsonContent = editor.getJSON();
+      const parsedContent = parseEditorContent(jsonContent);
+
+      onPersist(parsedContent);
```

Comment:
In order to minimize the number of times we do these time consuming calls, we might want to do these lazily:

```suggestion
      onPersist(() => {
        const jsonContent = editor.getJSON();
        const parsedContent = parseEditorContent(jsonContent);
      });
```

Reply:
See my previous comment, is it necessary to call onPersist at each change ?